### DISCUSSION TITLE
How to register local components (in a page's folder)

### DISCUSSION THREAD

**Discussion by saifobeidat** [nuxt/nuxt#22165] [Questions] [answered]
In Nuxt2, we used to put components in the same page's folder (Group by domain).. this way has several advantages, so we decided to stick with it while upgrading to Nuxt 3..

Example:
![image](https://github.com/nuxt/nuxt/assets/24841143/4f739282-1c7a-4c4c-b8dd-0dde0e221512)
as you see we prefix the component name by `-`.. so we make sure Nuxt deal with it as a page..

So while the upgrading process, we noticed that Nuxt doesn't detect any code change for those ignored components by `-` .. you need to kill the terminal each time you make a change.. (probably because it's ignored) 

Our project is big, we're trying to do minimal changes.. so any chance that we can make this work in Nuxt3?

**Comment by warflash** [comment, marked as answer]
Perhaps this post helps you get on track for finding a suitable config: https://deltener.com/blog/nuxt-enterprise-patterns-component-management/

**Reply by saifobeidat** [reply]
Nice!! Looks that's what i want, i going to look at it

Thank you 🙌

**Reply by saifobeidat** [reply]
@warflash works as expected! thank you
